llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -970,6 +984,21 @@ defset list<AMDGPUImageDimIntrinsic> AMDGPUImageDimIntrinsics = {
         AMDGPUImageDMaskIntrinsic;
   }
 
+  multiclass AMDGPUImageDimSampleNoReturnDims<string opmod,
+                                      AMDGPUSampleVariant sample> {
+    foreach dim = AMDGPUDims.NoMsaa in {
+      def !strconcat(NAME, "_", dim.Name) : AMDGPUImageDimIntrinsic<
```

Comment:
If it's the same signature as the existing cases, can you just overload the return type to void?

Reply:
The attributes are different, so I don't think they can share signatures?

Reply:
You could rely on callsite attributes but that's a big hassle